benchcheck_test.go

```diff
@@ -72,6 +80,33 @@ func TestGetModule(t *testing.T) {
 			if !fileinfo.IsDir() {
 				t.Fatalf("want %q to be a dir, details : %v", modDir, fileinfo)
 			}
+
+			if test.wantVersion == "" {
+				test.wantVersion = test.moduleVersion
+			}
+
+			gotModuleVersion := getModuleVersion(t, modDir)
+			if gotModuleVersion != test.wantVersion {
+				t.Fatalf(
+					"got version %q from mod dir %q, wanted %q",
+					gotModuleVersion,
+					modDir,
+					test.wantVersion,
+				)
+			}
 		})
 	}
 }
+
+func getModuleVersion(t *testing.T, modDir string) string {
+	// This function abuses private details abou the modDir,
+	// more specifically, the fact it is a Go cache dir.
+	// But it is at least decoupled from specific version systems.
+	// We couple on whatever interfaces Go provides.
```

Comment:
I would remove this last paragraph or move it to the top of the file docs, because being here implies go cache directory is an interface or something designed to be.

Reply:
It is an interface, exporting the cache as a HTTP service works exactly because the paths are part of a interface/protocol

Reply:
now that I think about it, it is not a private detail at all, it is part of the module/proxy protocol. Although I never read anything explicit about it, like a Go backward compatibility promisse or something, it is expected that exporting local files as HTTP will work, since the FS matches the REST API. Well will remove this comment too, they are not helping as much as I hoped.

Reply:
It makes sense. I love fs interfaces but this is odd on Linux. If they can hold this guarantee, kudos to them =)

Reply:
I'm almost sure they intend to, they don't talk explicitely about the mapping with FS, but the HTTP protocol itself is already established/documented and does have the same compatibility guarantee that Go provides AFAIK (module support is an essential part of the ecosystem now), so if it is compatible with the local FS now it will most likely keep being compatible.